src/forms/mod.rs

```diff
@@ -0,0 +1,153 @@
+//! Forms within your terminal
+use std::{any::Any, collections::HashMap};
+
+use crate::{events::Event, rect_style::BorderStyle, screen::Screen};
+
+mod choices;
+pub mod constraints;
+mod form;
+mod text;
+
+pub use choices::Checkbox;
+pub use choices::Radio;
+use crossterm::style::Color;
+pub use form::Form;
+pub use text::HiddenText;
+pub use text::Text;
+
+use self::constraints::FormConstraint;
+
+/// Helper trait to allow downcasting FormFields
+pub trait ToAny {
+    fn to_any(self) -> Box<dyn Any>;
+}
+impl<T: 'static> ToAny for T {
+    fn to_any(self) -> Box<dyn Any> {
+        Box::new(self)
+    }
+}
+
+/// Necessary functions to build a Form Field
+pub trait FormField: ToAny {
+    /// Base function to allow building the Field programmatically (e.g. in Forms)
+    ///
+    /// This function is separate to `new` to allow Fields to define custom constructors
+    fn make(w: u32, options: FormOptions) -> Self
+    where
+        Self: Sized;
+
+    /// Reset the field
+    fn reset(&mut self);
+
+    /// Get the width of the field
+    fn get_width(&self) -> u32;
+    /// Get the height of the field
+    fn get_height(&self) -> u32;
+    fn get_min_height(&self) -> u32 {
+        1
+    }
+    /// Resize (if possible) the field
+    fn resize(&mut self, w: u32, h: u32);
+
+    /// This function is the heart of FormFields : it allows the form to handle itself by passing a ConsoleEngine Event to it.
+    fn handle_event(&mut self, event: &Event);
+
+    /// Set the active state of a field (if applicable)
+    fn set_active(&mut self, active: bool);
+    /// Checks if the state of a field is active
+    fn is_active(&self) -> bool;
+
+    /// Allow the field to validate its content by itself.
+    ///
+    /// Make sure to call [self_validate](#method.self_validate) in it,
+    /// so the Validation Constraints provided by FormOptions will be automatically checked against the output of the field
+    fn validate(&self, validation_result: &mut FormValidationResult);
+    /// Get the output of the field
+    fn get_output(&self) -> FormValue;
+
+    /// Sets the options of the Field
+    fn set_options(&mut self, options: FormOptions);
+    /// Gets the options of the Field
+    fn get_options(&self) -> &FormOptions;
+    /// Tell if we should display the label externally (some fields may want to display it themselves like buttons)
+    fn display_label(&self) -> bool {
```

Comment:
`display_label` sounds like an action (like `draw` or `validate`) but it doesn't actually display the label. I'd suggest renaming it, to something like `should_display_label` (or probably something else).